services/polly/src/main/java/software/amazon/awssdk/services/polly/internal/presigner/DefaultPollyPresigner.java

```diff
@@ -212,11 +218,16 @@ private ExecutionAttributes createExecutionAttributes(PresignRequest presignRequ
                 .putAttribute(PRESIGNER_EXPIRATION, signatureExpiration);
     }
 
-    private AwsCredentialsProvider resolveCredentialsProvider(PollyRequest request) {
-        return request.overrideConfiguration().flatMap(AwsRequestOverrideConfiguration::credentialsProvider)
+    private IdentityProvider<? extends AwsCredentialsIdentity> resolveCredentialsProvider(PollyRequest request) {
+        return request.overrideConfiguration().flatMap(AwsRequestOverrideConfiguration::credentialsIdentityProvider)
                 .orElse(credentialsProvider());
     }
 
+    private AwsCredentials resolveCredentials(IdentityProvider<? extends AwsCredentialsIdentity> credentialsProvider) {
+        AwsCredentialsIdentity credentials = credentialsProvider.resolveIdentity().join();
+        return CredentialUtils.toCredentials(credentials);
+    }
```

Comment:
Can we return and use AwsCredentialsIdentity directly instead of AwsCredentials?